src/main/java/HttpServer/ResponseWriter.java

```diff
@@ -24,37 +21,42 @@ public void write() throws IOException {
         System.out.println("\nWriting response:");
         System.out.println("--------------------------");
 
-        String statusLine = computeStatusLine(request);
-        writeLine(statusLine);
+        Response basicResponse = respondWithStatus(request);
 
-        String contentType = "Content-Type: text/html";
-        writeLine(contentType);
+        String statusLine = basicResponse.statusLine();
+        writeLine(statusLine);
 
-        String contentLength = "Content-Length: 0";
-        writeLine(contentLength);
+        if (basicResponse.containsKey("Content-Type")) {
```

Comment:
There is a lot going on in this function. I think more responsibility could be pushed to the `Response` domain object. For example, what happens if you have multiple headers? What if you have a user defined header that your core application doesn't know about? A more extensible approach may be to have the response have a method to return all the headers, and just loop through them here.

Reply:
:heavy_check_mark: